src/v/utils/ema.h

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#pragma once
+#include <chrono>
+#include <vector>
+
+/*
+ * exponential weighted moving average
+ *
+ * 1. Create smoothing factor, initial average, window count
+ * 2. Update the current window by calling `update`
+ * 3. Call `tick` to advance to the next time window
+ *
+ * The implementation does not account for missed ticks, so it is best to drive
+ * ticks with a timer which is often convenient when used in some control algo.
+ * This has the added advantage that high frequency updates are cheaper because
+ * they do not need to test for and initialize new time windows.
+ *
+ * TODO:
+ *  - Generalize this on the type being counted. The structure of this ewma
+ *  utility is generic, but the implementation assumes that it is parameterized
+ *  on std::chrono durations.
+ */
+template<typename Clock = std::chrono::high_resolution_clock>
+class expo_moving_average {
+public:
+    using duration = typename Clock::duration;
+
+    // clamp to zero below this threshold
+    static constexpr double zero = 0.001;
```

Comment:
zero-clamp? 

also question, can we get to the same math without using floating point arithmetic which is expensive

Reply:
probably. but we can reduce sampling down to once per window period which is about 7 seconds. so its like 20 odd flops every 7 seconds.

Reply:
fixed